File: bumperbot_python_examples/bumperbot_python_examples/simple_service_server.py
import rclpy
from rclpy.node import Node
from bumperbot_msgs.srv import AddTwoInts



class SimpleServiceServer(Node):
    def __init__(self):
        super().__init__("simple_service_server")
        self_server_ = self.create_service(AddTwoInts, "add_two_ints", self.AddCallback)

        self.get_logger().info(f"add two ints server has been started!")

    def AddCallback(self, req, res):
        self.get_logger().info(f"request to add a:{req.a} + b:{req.b}")
        res.sum = req.a + req.b
        self.get_logger().info(f"response is {res.sum}")
        
        return res
    

def main():
    rclpy.init()
    node = SimpleServiceServer()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: bumperbot_cpp_examples/src/simple_service_server.cpp
#include <rclcpp/rclcpp.hpp>
#include <bumperbot_msgs/srv/add_two_ints.hpp>

using namespace std::placeholders;

class SimpleServiceServer: public rclcpp::Node
{
    public:
    SimpleServiceServer(): Node("simple_service_server")
    {
        server_ = create_service<bumperbot_msgs::srv::AddTwoInts>("add_two_ints", std::bind(&SimpleServiceServer::addCallback, this, _1, _2));
        RCLCPP_INFO_STREAM(get_logger(), "simple_service_server has been started!");

    }

    private:
    rclcpp::Service<bumperbot_msgs::srv::AddTwoInts>::SharedPtr server_;
    void addCallback(
        std::shared_ptr<bumperbot_msgs::srv::AddTwoInts::Request> req, 
        std::shared_ptr<bumperbot_msgs::srv::AddTwoInts::Response> res
    )
    {
        RCLCPP_INFO_STREAM(get_logger(), "request a:" << req->a << " + b:" << req->b);
        res->sum = req->a + req->b;
        RCLCPP_INFO_STREAM(get_logger(), "response sum:" << res->sum);
    
    }


};

int main(int argc, char* argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleServiceServer>();
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}


